workerpool.go

```diff
@@ -200,9 +202,27 @@ func (wp *workerPool) release(ch *workerChan) bool {
 	return true
 }
 
+func (wp *workerPool) workerDone() {
+	wp.lock.Lock()
+	wp.workersCount--
+	wp.lock.Unlock()
+}
+
 func (wp *workerPool) workerFunc(ch *workerChan) {
```

Comment:
I don't really like that this change will silently affect users with PanicHandler == nil as well.

`if wp.PanicHandler != nil` for == nil would at least execute redundant `JZ`, and the `if wp.PanicHandler == nil` below implies something like

```asm
JNZ FUNCRET;
JMP WORKER_DONE;
FUNCRET:
```

and that's all while the user doesn't even use this feature.

while I understand what (*wp).workerDone() is easier to read in the context of this feature, I'm not sure that this feature is really required, or at least that this implementation is optimal for both users who want the PanicHandler and who don't.

Again, I use fasthttp for several years now and I still didn't find a case when fasthttp would panic.

I'd like to hear some more thoughts about this PR.

Reply:
I also think it's better if we just handle panics from `Response.SetBodyStream` and `Response.SetBodyStreamWriter` and use `s.logger().Printf("...` and return without writing anything more in the response as we don't know what's written already.

For people who want more control it's not that hard to wrap the reader and handle these panics themselves.

This way we don't expose any extra API again.

Reply:
I agree with the above. 

Moreover, https://github.com/valyala/fasthttp/pull/687 is definitely a step in the right direction for us. So closing this one.